### PR TITLE
homework 3

### PR DESCRIPTION
**Pull request by yspryn**

### CODE REVIEW

File: tests/yspryn/HW2/conftest.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import pytest
+from tests.yspryn.HW2.locators import *
```
Comment (fkudinov-mn) [inline tests/yspryn/HW2/conftest.py:2]: I would not suggest wild card

Comment (fkudinov-mn) [reply tests/yspryn/HW2/conftest.py:2]: from tests.yspryn.HW2 import locators

session.find_element(*locators.TXT_LOGIN_INPUT).send_keys(LOGIN)


or


from tests.yspryn.HW2 import locators as L
session.find_element(*L.TXT_LOGIN_INPUT).send_keys(LOGIN)

File: tests/yspryn/HW2/locators.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+from selenium.webdriver.common.by import By
+
+
+"""Login page locators"""
+# before-> (By.ID, "user-name")
+TXT_LOGIN_INPUT = (By.CSS_SELECTOR, "input.form_input.input_error#user-name")
+# before-> (By.ID, "password")
+TXT_PASSWORD_INPUT = (By.CSS_SELECTOR, "input.form_input.input_error#password[type='password']")
+# before-> (By.ID, "login-button")
+BTN_SUBMIT = (By.XPATH, "//input[@id='login-button']")
```
Comment (fkudinov-mn) [inline tests/yspryn/HW2/locators.py:10]: better to group into classes


class LoginPage:
        """Login page locators"""

        # before-> (By.ID, "user-name")
        TXT_LOGIN_INPUT = (By.CSS_SELECTOR, "input.form_input.input_error#user-name")
        # before-> (By.ID, "password")
        TXT_PASSWORD_INPUT = (By.CSS_SELECTOR, "input.form_input.input_error#password[type='password']")
        # before-> (By.ID, "login-button")
        BTN_SUBMIT = (By.XPATH, "//input[@id='login-button']")

File: tests/yspryn/HW2/locators.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+from selenium.webdriver.common.by import By
+
+
+"""Login page locators"""
+# before-> (By.ID, "user-name")
+TXT_LOGIN_INPUT = (By.CSS_SELECTOR, "input.form_input.input_error#user-name")
```
Comment (fkudinov-mn) [inline tests/yspryn/HW2/locators.py:6]: id is unique enough 

(By.CSS_SELECTOR, "#user-name")

Comment (fkudinov-mn) [reply tests/yspryn/HW2/locators.py:6]: (By.ID, "some_id")  ->     (By.CSS_SELECTOR, "#some_id") -> (By.XPATH, "//*[@id='some_id']")

File: tests/yspryn/HW2/locators.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+# before-> (By.ID, "password")
+TXT_PASSWORD_INPUT = (By.CSS_SELECTOR, "input.form_input.input_error#password[type='password']")
+# before-> (By.ID, "login-button")
+BTN_SUBMIT = (By.XPATH, "//input[@id='login-button']")
+
+
+"""Landing page"""
+# before-> (By.CSS_SELECTOR, ".inventory_list .inventory_item")
+TABLE_PRODUCT_ITEMS = (By.XPATH, "//*[@class='inventory_list']/*[@class='inventory_item']")
```
Comment (fkudinov-mn) [inline tests/yspryn/HW2/locators.py:15]: use 'contains', you have no guarantee that nobody adds additional class name to element

"//*[contains(@class, 'inventory_list')]//*[contains(@class, 'inventory_item')]"

File: tests/yspryn/HW2/locators.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+# before-> (By.XPATH, ".//*[@class='pricebar']//button")
+BTN_ADD_TO_CART = (By.CSS_SELECTOR, "div>button.btn_inventory.btn_small.btn.btn_primary")
```
Comment (fkudinov-mn) [inline tests/yspryn/HW2/locators.py:17]: wrong translation,

(By.CSS_SELECTOR, ".pricebar button")

File: tests/yspryn/HW2/locators.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+# before-> (By.ID, "shopping_cart_container")
+BTN_CART_LOCATE = (By.XPATH, "//div[@id='shopping_cart_container']")
```
Comment (fkudinov-mn) [inline tests/yspryn/HW2/locators.py:19]: "//*[@id='shopping_cart_container']"

File: tests/yspryn/HW2/locators.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+# before-> (By.XPATH, ".//*[contains(@class, 'shopping_cart_badge')]")
+TXT_CART_BADGE = (By.CSS_SELECTOR, "span.shopping_cart_badge")
```
Comment (fkudinov-mn) [inline tests/yspryn/HW2/locators.py:21]: ".shopping_cart_badge"

File: tests/yspryn/HW2/locators.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+
+"""Cart page"""
+# before-> (By.CSS_SELECTOR, ".cart_list .cart_item")
+TABLE_ITEMS_IN_CART = (By.XPATH, "//*[@class='cart_list']/*[@class='cart_item']")
```
Comment (fkudinov-mn) [inline tests/yspryn/HW2/locators.py:26]: use contains

File: tests/yspryn/HW2/locators.py [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+# before-> (By.CSS_SELECTOR, ".btn_secondary.btn_small.cart_button")
+BTN_REMOVE_FROM_CART = (By.XPATH, "//*[@class='cart_item']//button")
```
Comment (fkudinov-mn) [inline tests/yspryn/HW2/locators.py:28]: wrong translation from xpath

File: tests/yspryn/HW2/test_chrome_site.py [hunk 10]
```diff
@@ -1,6 +1,6 @@
 import pytest
+from tests.yspryn.HW2.locators import *
```
Comment (fkudinov-mn) [inline tests/yspryn/HW2/test_chrome_site.py:2]: do not use wild card

**Review by kudinov-fedor — APPROVED** [review]

**Pull request merged**